src/components/EditorsColumn.jsx

```diff
@@ -1,70 +1,164 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
+import {DraggableCore} from 'react-draggable';
+import bindAll from 'lodash/bindAll';
 import isEmpty from 'lodash/isEmpty';
 import includes from 'lodash/includes';
 import partial from 'lodash/partial';
 
 import EditorContainer from './EditorContainer';
 import Editor from './Editor';
 
+
+const DIVIDER_HEIGHT = 4;
+const EDITOR_MIN_HEIGHT = 85;
+
+
 function allErrorsFor(language, errors, runtimeErrors) {
   if (language === 'javascript') {
     return errors.javascript.items.concat(runtimeErrors);
   }
   return errors[language].items;
 }
 
-export default function EditorsColumn({
-  currentProject,
-  errors,
-  onComponentMinimize,
-  onEditorInput,
-  onRequestedLineFocused,
-  runtimeErrors,
-  ui,
-}) {
-  const editors = [];
-  ['html', 'css', 'javascript'].forEach((language) => {
-    if (includes(ui.minimizedComponents, `editor.${language}`)) {
-      return;
+function isSecondDividerPushingUpFirstDivider(height0, height1, deltaY) {
+  return height0 > EDITOR_MIN_HEIGHT &&
+    height1 === EDITOR_MIN_HEIGHT - DIVIDER_HEIGHT &&
+    deltaY < 0;
+}
+
+function adjustY(height2, lastY, deltaY) {
+  const maxDeltaY = height2 - EDITOR_MIN_HEIGHT;
+  return lastY + Math.min(deltaY, maxDeltaY);
+}
+
+function extractNodeHeights(editorRefs) {
+  const nodes = editorRefs.filter(Boolean).map(ReactDOM.findDOMNode);
+  const [
+    {offsetHeight: height0},
+    {offsetHeight: height1},
+  ] = nodes;
+  return [
+    height0,
+    height1,
+    (nodes.length === 3) ? nodes[2].offsetHeight : null,
+  ];
```

Comment:
I think it would feel better to just `map` over the nodes? Do we need there to always be three elements?